pyiceberg/table/__init__.py

```diff
@@ -988,11 +988,15 @@ def commit_transaction(self) -> Table:
         Returns:
             The table with the updates applied.
         """
-        self._requirements = (AssertCreate(),)
-        self._table._do_commit(  # pylint: disable=W0212
-            updates=self._updates,
-            requirements=self._requirements,
-        )
+        if len(self._updates) > 0:
+            self._table._do_commit(  # pylint: disable=W0212
+                updates=self._updates,
+                requirements=(AssertCreate(),),
+            )
+
+        self._updates = ()
+        self._requirements = ()
```

Comment:
nit: can we use `super().commit_transaction` here? 

I want to make sure there's a way for future subclass of `Transaction` to also clear the updates and requirements